tests/components/matter/test_binary_sensor.py

```diff
@@ -344,3 +345,30 @@ async def test_water_valve(
     state = hass.states.get("binary_sensor.valve_valve_leaking")
     assert state
     assert state.state == "on"
+
+
+@pytest.mark.parametrize("node_fixture", ["thermostat"])
+async def test_thermostat_occupancy(
+    hass: HomeAssistant,
+    matter_client: MagicMock,
+    matter_node: MatterNode,
+) -> None:
+    """Test thermostat occupancy."""
+    state = hass.states.get("binary_sensor.longan_link_hvac_occupancy")
+    assert state
+    assert state.state == "on"
+
+    # Test Occupancy attribute change
+    occupancy_attribute = clusters.Thermostat.Attributes.Occupancy
+
+    set_node_attribute(
+        matter_node,
+        1,
+        occupancy_attribute.cluster_id,
+        occupancy_attribute.attribute_id,
+        0,
+    )
+    await trigger_subscription_callback(hass, matter_client)
+    state = hass.states.get("binary_sensor.longan_link_hvac_occupancy")
+    assert state
+    assert state.state == "off"
```

Comment:
very small nitpick: could use an empty line after awaiting `trigger_subscription_callback` to match some/most of the other binary sensor tests

```suggestion
    await trigger_subscription_callback(hass, matter_client)

    state = hass.states.get("binary_sensor.longan_link_hvac_occupancy")
    assert state
    assert state.state == "off"
```